raiden/transfer/mediated_transfer/events.py

```diff
@@ -66,18 +70,31 @@ class SendBalanceProof(Event):
     Note:
         This event has a dual role, it serves as a synchronization and as
         balance-proof for the netting channel smart contract.
+
+        Nodes need to keep the last known merkle root synchronized. This is
+        required by the receiving end of a transfer in order to properly
+        validate. The rule is "only the party that owns the current payment
+        channel may change it" (remember that a netting channel is composed of
+        two uni-directional channels), as a consequence the merkle root is only
+        updated by the receiver once a balance proof message is received.
     """
-    def __init__(self, identifier, target):
+    def __init__(self, identifier, channel_address, token, receiver, secret):
         self.identifier = identifier
-        self.target = target
+        self.channel_address = channel_address
+        self.token = token
+        self.receiver = receiver
+
+        # XXX: Secret is not required for the balance proof to dispatch the message
+        self.secret = secret
 
 
 class SendSecretRequest(Event):
     """ Event used by a target node to request the secret from the initiator. """
-    def __init__(self, identifier, amount, hashlock):
+    def __init__(self, identifier, amount, hashlock, receiver):
```

Comment:
Yes, using receiver to be consistent with the other Send events, do you want me to change it?